WickedEngine/wiGraphicsDevice_Vulkan.cpp

```diff
@@ -2767,14 +2767,45 @@ using namespace vulkan_internal;
 		allocationhandler->instance = instance;
 
 		// Initialize Vulkan Memory Allocator helper:
+		VmaVulkanFunctions vma_vulkan_func{};
+		vma_vulkan_func.vkGetInstanceProcAddr = vkGetInstanceProcAddr;
+		vma_vulkan_func.vkGetDeviceProcAddr = vkGetDeviceProcAddr;
+		vma_vulkan_func.vkGetPhysicalDeviceProperties = vkGetPhysicalDeviceProperties;
+		vma_vulkan_func.vkGetPhysicalDeviceMemoryProperties = vkGetPhysicalDeviceMemoryProperties;
+		vma_vulkan_func.vkAllocateMemory = vkAllocateMemory;
+		vma_vulkan_func.vkFreeMemory = vkFreeMemory;
+		vma_vulkan_func.vkMapMemory = vkMapMemory;
+		vma_vulkan_func.vkUnmapMemory = vkUnmapMemory;
+		vma_vulkan_func.vkFlushMappedMemoryRanges = vkFlushMappedMemoryRanges;
+		vma_vulkan_func.vkInvalidateMappedMemoryRanges = vkInvalidateMappedMemoryRanges;
+		vma_vulkan_func.vkBindBufferMemory = vkBindBufferMemory;
+		vma_vulkan_func.vkBindImageMemory = vkBindImageMemory;
+		vma_vulkan_func.vkGetBufferMemoryRequirements = vkGetBufferMemoryRequirements;
+		vma_vulkan_func.vkGetImageMemoryRequirements = vkGetImageMemoryRequirements;
+		vma_vulkan_func.vkCreateBuffer = vkCreateBuffer;
+		vma_vulkan_func.vkDestroyBuffer = vkDestroyBuffer;
+		vma_vulkan_func.vkCreateImage = vkCreateImage;
+		vma_vulkan_func.vkDestroyImage = vkDestroyImage;
+		vma_vulkan_func.vkCmdCopyBuffer = vkCmdCopyBuffer;
+
 		VmaAllocatorCreateInfo allocatorInfo = {};
 		allocatorInfo.physicalDevice = physicalDevice;
 		allocatorInfo.device = device;
 		allocatorInfo.instance = instance;
+
+		// Core in 1.1
+		allocatorInfo.flags = VMA_ALLOCATOR_CREATE_KHR_DEDICATED_ALLOCATION_BIT | VMA_ALLOCATOR_CREATE_KHR_BIND_MEMORY2_BIT;
```

Comment:
Why do we need these flags now?

Reply:
VMA can make use of dedicated allocation and bind memory extensions which are core in 1.1, if you want we can disable them.

Reply:
It seems that there is equivalent functionality in core, but these are using KHR in their names. Does VMA use the core or KHR extensions? Because if it uses extension, I think it needs to be enabled separately.

Reply:
Not sure, I've baaed the implementation on vulkan samples: https://github.com/KhronosGroup/Vulkan-Samples/blob/master/framework/core/device.cpp#L226 feel free to change it